### ISSUE TITLE
Tensorflow Support

### ISSUE THREAD

**Issue by LucasSloan** [HazyResearch/flash-attention#35]
Are you aware of any efforts to support this in Tensorflow?

**Comment by tridao** [comment]
We've been talking to TPU folks about this, but we don't have anything concrete yet. It's a great opportunity for collaboration!

There's also an implementation in Jax, by @lucidrains: https://github.com/lucidrains/flash-attention-jax

**Comment by lucidrains** [comment]
@LucasSloan i've heard rumors that certain XLA compiler folks are looking for a way to make it work. but it is probably better to ask James Bradbury or someone more knowledgeable

**Comment by lucidrains** [comment]
@LucasSloan right now flash attention is only feasible on CUDA, as you have fine control over memory access patterns

**Comment by LucasSloan** [comment]
Yep, I use Tensorflow with GPUs.  Presumably, one could hook the CUDA kernels here into tensorflow.

**Comment by tridao** [comment]
That's a great idea!
I'm not experienced in hooking custom CUDA code into Tensorflow, but I think it'll make a great collaboration. 
Is there someone we should reach out to regarding this?

**Comment by oliverlwang** [comment]
hi guys, I haven't heard any updates. Is there any progress on this issue?

**Comment by fuhailin** [comment]
TF version of flash attention is coming: https://github.com/intelligent-machine-learning/dlrover/tree/master/tfplus/tfplus/flash_attn

**Comment by gravytrain777** [comment]
In case anyone still using tensorflow in 2026, you can check out my repo here [https://github.com/gravytrain777/tf-custom-flash-attention](url)
